thrifty-schema/src/main/java/com/microsoft/thrifty/schema/Program.java

```diff
@@ -240,14 +242,52 @@ void loadIncludedPrograms(Loader loader, Set<Program> visited) throws IOExceptio
         this.includedPrograms = includes.build();
 
         LinkedHashMap<String, Named> symbolMap = new LinkedHashMap<>();
-        for (Named named : names()) {
-            Named oldValue = symbolMap.put(named.name(), named);
+
+        // Some type-resolution subtlety eludes me.  I'd have thought that
+        // Iterable<EnumType> is castable to Iterable<Named> (inheritance),
+        // but the IDE claims otherwise.  So, instead create a temp list where each list's
+        // data will be copied into
+        List<Named> namedList = new ArrayList<>();
```

Comment:
I agree that we need to separate duplicate-symbol-detection of constants from that of types.  That said, this seems to be too loose - we will only detect name clashes within one particular subtype of `Named`; a `struct` and a `union` with the same name is definitely not legal, but will pass validation this way.

We ought to just do two passes on `names()`, one where we check constants and another where we check non-constants.

On a deeper level this change raises a troubling issues - the way we store and query symbols is now shown to be broken.  A name could refer to either a constant or a type, and we currently don't allow that.  This requires some thought...